core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -42,6 +42,8 @@ private[spark] abstract class SparkFunSuite extends FunSuite with Logging {
       test()
     } finally {
       logInfo(s"\n\n===== FINISHED $shortSuiteName: '$testName' =====\n")
+      // Avoid leaking map entries in tests that use accumulators without SparkContext
+      Accumulators.clear()
```

Comment:
We want to isolate state across each test. Otherwise we might get an accumulator that's registered in a previous test and that's confusing if it fails.